src/WinRT.Runtime/ComWrappersSupport.cs

```diff
@@ -23,16 +22,25 @@
 
 namespace WinRT
 {
+    internal static class KeyValuePairHelper
+    {
+        internal readonly static ConcurrentDictionary<Type, ComInterfaceEntry> KeyValuePairCCW = new();
+
+        internal static void TryAddKeyValuePairCCW(Type keyValuePairType, Guid iid, IntPtr abiToProjectionVftablePtr)
+        {
+            KeyValuePairCCW.TryAdd(keyValuePairType, new ComInterfaceEntry { IID = iid, Vtable = abiToProjectionVftablePtr });
+        }
+    }
+
 #if EMBED
     internal
 #else
     public 
 #endif
     static partial class ComWrappersSupport

```

Comment:
I see a few calls to `GuidGenerator` in `ProvideIReferenceArray` below, for constructed generic types resolved over specific type arguments. Would it be possible in theory to just hardcode those GUIDs instead? 🤔

Reply:
These probably can be hard coded.  Will address this in a future PR.